False positive `implicit_getter` violation when mentioning word "set" in a comment in custom getter of a read-write var

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Mentioning word "set" (without parentheses) in a **comment** in a `get` of a read-write var with both get and set defined triggers SwiftLint to think that this is a read-only property.

<img width="819" alt="image" src="https://user-images.githubusercontent.com/217368/77595323-1560f380-6eb6-11ea-838f-a6277fb23a6f.png">

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint --path Test.swift --no-cache --enable-all-rules
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths Test.swift
Linting 'Test.swift' (1/1)
/Users/yas/code/caremobile/Test.swift:3:9: warning: Implicit Getter Violation: Computed read-only properties should avoid using the get keyword. (implicit_getter)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint 0.39.1
* Installation method used: Homebrew
* Paste your configuration file:

```yml
excluded:
  - Pods

opt_in_rules:
  - array_init
  - contains_over_first_not_nil
  - collection_alignment
  - contains_over_filter_count
  - empty_count
  - empty_collection_literal
  - overridden_super_call
  - sorted_first_last
  - yoda_condition

disabled_rules:
  - cyclomatic_complexity
  - file_length
  - force_cast
  - force_try
  - function_body_length
  - function_parameter_count
  - identifier_name
  - large_tuple
  - line_length
  - multiple_closures_with_trailing_closure
  - nesting # consider
  - notification_center_detachment
  - operator_whitespace
  - todo # fix
  - type_body_length
  - type_name # consider
  - weak_delegate # fix

statement_position:
  statement_mode: uncuddled_else
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?No

* Which Xcode version are you using (check `xcodebuild -version`)?
Xcode 11.4
Build version 11E146

* Do you have a sample that shows the issue? 
Yes:

**Test.swift**
```swift
extension Test {
    var foo: Bool {
        get {
            bar?.boolValue ?? true // Comment mentioning word set which triggers violation
        }
        set {
            bar = NSNumber(value: newValue as Bool)
        }
    }
}
```

If I don't mention word "set" in the comment then it seem to work fine.
I don't recall seeing this issues prior to updating to Xcode 11.4.

Hope it helps.

I found another false-positive. When the `set` is before the `get` you get warning. If it is the other way round then everything is all right. I believe another rule could be created to formalize the order of get-set.

```swift
extension Float {
    var clamped: Float {
        set {
            self = min(1, max(0, newValue))
        }
        get {
            min(1, max(0, self))
        }
    }
}

```

Command:

```
swiftlint
```

Output:

```
Linting Swift files at paths 
Linting 'lint.swift' (1/1)
/Users/kaspar/Downloads/swiftlint-issue/lint.swift:6:9: warning: Implicit Getter Violation: Computed read-only properties should avoid using the get keyword. (implicit_getter)
Done linting! Found 1 violation, 0 serious in 1 file.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,7 +21,10 @@
 
 #### Bug Fixes
 
-* None.
+* Fix more false positive in `implicit_getter` rule in extensions when using
+  Swift 5.2.  
+  [Marcelo Fabri](https://github.com/marcelofabri)
+  [#3149](https://github.com/realm/SwiftLint/issues/3074)
 
 ## 0.39.1: The Laundromat has a Rotating Door
 

--- a/Source/SwiftLintFramework/Rules/Style/ImplicitGetterRule.swift
+++ b/Source/SwiftLintFramework/Rules/Style/ImplicitGetterRule.swift
@@ -16,7 +16,7 @@ public struct ImplicitGetterRule: ConfigurationProviderRule, AutomaticTestableRu
     )
 
     public func validate(file: SwiftLintFile) -> [StyleViolation] {
-        let getTokens = findComputedPropertyToken(keyword: "get", file: file)
+        let getTokens = findGetTokens(file: file)
 
         let violatingLocations = getTokens.compactMap { token -> (ByteCount, SwiftDeclarationKind?)? in
             // the last element is the deepest structure
@@ -31,11 +31,11 @@ public struct ImplicitGetterRule: ConfigurationProviderRule, AutomaticTestableRu
                     return nil
                 }
             } else {
-                guard let range = dict.bodyByteRange.map(file.stringView.byteRangeToNSRange) else {
+                guard let range = dict.byteRange.map(file.stringView.byteRangeToNSRange) else {
                     return nil
                 }
 
-                let setTokens = findComputedPropertyToken(keyword: "set", file: file, range: range)
+                let setTokens = findSetTokens(file: file, range: range)
                 let hasSetToken = setTokens.contains { token in
                     // the last element is the deepest structure
                     guard let setDict = declarations(forByteOffset: token.offset,
@@ -68,15 +68,27 @@ public struct ImplicitGetterRule: ConfigurationProviderRule, AutomaticTestableRu
         }
     }
 
-    private func findComputedPropertyToken(keyword: String, file: SwiftLintFile,
-                                           range: NSRange? = nil) -> [SwiftLintSyntaxToken] {
-        let pattern = "\\{[^\\{]*?\\s+\(keyword)\\b"
+    private func findGetTokens(file: SwiftLintFile) -> [SwiftLintSyntaxToken] {
+        let pattern = "\\{[^\\{]*?\\s+get\\b"
         let attributesKinds: Set<SyntaxKind> = [.attributeBuiltin, .attributeID]
-        return file.rangesAndTokens(matching: pattern, range: range).compactMap { _, tokens in
+        return file.rangesAndTokens(matching: pattern).compactMap { _, tokens in
             let kinds = tokens.kinds
             guard let token = tokens.last,
                 token.kind == .keyword,
                 attributesKinds.isDisjoint(with: kinds) else {
+                    return nil
+            }
+
+            return token
+        }
+    }
+
+    private func findSetTokens(file: SwiftLintFile, range: NSRange?) -> [SwiftLintSyntaxToken] {
+        let pattern = "\\bset\\b"
+        return file.rangesAndTokens(matching: pattern).compactMap { _, tokens in
+            guard tokens.count == 1,
+                let token = tokens.last,
+                token.kind == .keyword else {
                     return nil
             }
 

--- a/Source/SwiftLintFramework/Rules/Style/ImplicitGetterRuleExamples.swift
+++ b/Source/SwiftLintFramework/Rules/Style/ImplicitGetterRuleExamples.swift
@@ -1,3 +1,4 @@
+// swiftlint:disable:next type_body_length
 struct ImplicitGetterRuleExamples {
     static var nonTriggeringExamples: [Example] {
         let commonExamples = [
@@ -98,6 +99,42 @@ struct ImplicitGetterRuleExamples {
                 var bar: Bool {
                     get { _bar }
                     set(newValue) { self._bar = newValue }
+                }
+            }
+            """),
+            Example("""
+            extension Float {
+                var clamped: Float {
+                    set {
+                        self = min(1, max(0, newValue))
+                    }
+                    get {
+                        min(1, max(0, self))
+                    }
+                }
+            }
+            """),
+            Example("""
+            extension Reactive where Base: UITapGestureRecognizer {
+                var tapped: CocoaAction<Base>? {
+                    get {
+                        return associatedAction.withValue { $0.flatMap { $0.action } }
+                    }
+                    nonmutating set {
+                        setAction(newValue)
+                    }
+                }
+            }
+            """),
+            Example("""
+            extension Test {
+                var foo: Bool {
+                    get {
+                        bar?.boolValue ?? true // Comment mentioning word set which triggers violation
+                    }
+                    set {
+                        bar = NSNumber(value: newValue as Bool)
+                    }
                 }
             }
             """)

